perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -5091,8 +5091,16 @@ protected void initialize() throws InternalErrorException {
 		attr.setDisplayName("Last Synchronization timestamp");
 		attributes.add(attr);
 
+		if(perun.isThisPerunReadOnly()) log.debug("Loading attributes manager init in readOnly version.");
+
 		for(AttributeDefinition attribute : attributes) {
-			if(!checkAttributeExistsForInitialize(attribute)) createAttributeExistsForInitialize(attribute);
+			if(!checkAttributeExistsForInitialize(attribute)) {
+				if(perun.isThisPerunReadOnly()) {
+					log.error("One of required attributes not exists {}.", attribute);
```

Comment:
Maybe I'm missing it, but shouldn't we stop Perun startup here ? Because here we can't create attributes required by core functionality and they don't exists in DB.